src/test/java/org/jenkinsci/plugins/workflow/cps/CpsThreadDumpActionTest.java

```diff
@@ -40,11 +42,13 @@
     @Test public void doProgramDotXml() throws Exception {
         WorkflowJob p = r.createProject(WorkflowJob.class);
         p.setDefinition(new CpsFlowDefinition("node {semaphore 'wait'}", true));
-        WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+        final QueueTaskFuture<WorkflowRun> f = p.scheduleBuild2(0);
```

Comment:
It would be nice if `JenkinsRule` had a `waitAndAssertBuildStatus(Run)` method, or if `JenkinsRule.waitForCompletion` returned the `JenkinsRule`, since right now you either have to do what you did and create a variable for the `QueueTaskFuture`, or call `JenkinsRule.waitForCompletion` on its own line, both of which feel a little messy.

Reply:
I totally agree with you.